08-components-lab/src/02-numberguessing.jsx

```diff
@@ -11,7 +11,88 @@
  * To make the game interesting, computer should sometimes cheat and print wrong messages.
  */
 
+/**
+ * Lab 01:
+ *
+ * Implement a time converter component.
+ * Component should show 3 input fields labeled hours, minutes and seconds.
+ * Typing a value in either field updates the other two with corresponding values.
+ */
+
 import React from 'react';
 
-export default React.createClass({
-});
+export default class NumberGuessing extends React.Component {
+    static propTypes = {
+        initialNumber: React.PropTypes.number,
+        initialIsBigger: React.PropTypes.bool,
+
+    };
+
+    randomNumber() {
+        return Math.floor((Math.random() * 1001));
+    }
+
+    static defaultProps = {
+        initialNumber: Math.floor((Math.random() * 1001)),//this.randomNumber,
+        initialIsBigger: true
+    };
+
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            number: props.initialNumber,
+            isBigger: props.initialIsBigger
+        };
+    }
+
+    handleRandomGuess = (event) => {
+        var layOrTrue=Math.floor((Math.random() * 10));
+        if (event.target.value >= this.state.number) {
```

Comment:
Instead of using multiple times ev.target.value you can assign it to a const in the beginning of a method. Alternative is to use ES6 destructuring in arrow function arguments.
Consider this:

```
handleRandomGuess({ target: { value }) {
   //...
}
```